catroid/src/main/java/org/catrobat/catroid/embroidery/EmbroideryList.java

```diff
@@ -32,16 +32,25 @@
 	@Override
 	public synchronized boolean add(Object o) {
 		boolean status = false;
-		PointF point = (PointF) o;
-
-		if (super.isEmpty() || isNewStitchPoint(point)) {
+		if (o instanceof PointF) {
+			PointF point = (PointF) o;
+
+			if (super.isEmpty() || isNewStitchPoint(point)) {
+				status = super.add(new StitchPoint(point));
+			}
+		} else if (o instanceof StitchPoint) {
```

Comment:
stitchpoint should be a PointF and not have a PointF. if you want to have an ArrayList working this way, or youd have to remove the possibility to add PointF directly, and rather do the new StitchPoint(point) outside when youre trying to add a PointF
this whole EmbroideryList is not how one would normally deal with polymorphism in java.